packages/analytics-core/src/plugins/destination.ts

```diff
@@ -302,6 +302,14 @@ export class Destination implements DestinationPlugin {
       return;
     }
     const events = Array.from(this.queue.map((context) => context.event));
+
+    if (this.config.storageProvider.savedMaxCount && events.length > this.config.storageProvider.savedMaxCount) {
+      const deletedEvents = events.splice(0, events.length - this.config.storageProvider.savedMaxCount);
+      this.config.loggerProvider.error(
+        `Exceeded maximum saved events count. ${deletedEvents.length} events are dropped.`,
+      );
+    }
+
```

Comment:
imo the storage provider should enforce the limits and not its consumers if the config is set in storage provider

Reply:
Should we check in the storage, if the current events length is > savedMaxCount,  while trying to save the upcoming event, instead of doing some modification in the destination plugins?

Reply:
Move the logic to local storage rather than destination plugin